pkg/dl/dl.go

```diff
@@ -48,35 +49,80 @@ func New(name string, flags int) *DynamicLibrary {
 	}
 }
 
+func useLibDL(action func() error) error {
```

Comment:
I don't think `useLibDL` is a good name for this function. What about `withOSLock()` or `withSync()`?

Reply:
Renamed